helm/values.schema.json

```diff
@@ -0,0 +1,1710 @@
+                      "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEvtc/qpHtx7iUUj+rRHR99a8mnGni\nqiGkmUb9YpWWTS4YwlvwdmMDiGzcsHiDOYz6f88u2hCRF5GUCvyiZAKrsA==\n-----END PUBLIC KEY-----\n"
+                    }
+                  ]
+                }
+              ],
+              "if": {
+                "properties": {
+                  "type": {
+                    "const": "cosign"
+                  }
+                }
+              },
+              "then": {
+                "properties": {
+                  "trust_roots": {
+                    "$id": "#/properties/validators/items/anyOf/2/properties/trust_roots",
+                    "type": "array",
+                    "title": "The trust_roots schema for cosign validator",
+                    "description": "The trust root of the cosign validator. Currently, only public keys are supported as trust roots, but there may come other kinds of trust roots (e.g. mail addresses with OIDC flow).",
+                    "examples": [
+                      [
+                        {
+                          "name": "mykey",
+                          "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEvtc/qpHtx7iUUj+rRHR99a8mnGni\nqiGkmUb9YpWWTS4YwlvwdmMDiGzcsHiDOYz6f88u2hCRF5GUCvyiZAKrsA==\n-----END PUBLIC KEY-----\n"
+                        }
+                      ]
+                    ],
+                    "items": {
+                      "$id": "#/properties/validators/items/anyOf/2/properties/trust_roots",
+                      "anyOf": [
+                        {
+                          "$id": "#/properties/validators/items/anyOf/2/properties/trust_roots/anyOf/0",
+                          "type": "object",
+                          "title": "The first anyOf schema",
+                          "properties": {
+                            "name": {
+                              "$id": "#/properties/validators/items/anyOf/2/properties/trust_roots/anyOf/0/properties/name",
+                              "type": "string",
+                              "title": "The trust root's name schema",
+                              "description": "Name of the trust root",
+                              "examples": [
+                                "mykey"
+                              ]
+                            },
+                            "key": {
+                              "$id": "#/properties/validators/items/anyOf/2/properties/trust_roots/anyOf/0/properties/key",
+                              "type": "string",
+                              "title": "The trust root's key schema.",
+                              "pattern": "(?:-+BEGIN\\sPUBLIC\\sKEY[-]+)\n(?:(?:[A-Za-z0-9+\\/\\s])*={0,2})\n(?:-+END\\sPUBLIC\\sKEY[-]+)",
+                              "description": "Public ECDSA key used for verification of signatures on container images whose policy matches this trust root.",
```

Comment:
Is this ECDSA only? If so, since we include the cosign binaries, would we limit to ECDSA if they allowed ie RSAPSS?

Reply:
will generalize :-)